kernel/src/platform/watchdog.rs

```diff
@@ -0,0 +1,27 @@
+//! Interface for configuring a watchdog
+
+/// A trait for implementing a watchdog in the kernel.
+/// This trait is called from the `kernel_loop()` code to setup
+/// and maintain the watchdog timer.
+/// It is up to the specific `Chip` how it will handle watchdog interrupts.
+pub trait WatchDog {
+    /// This function must enable the watchdog timer and configure it to
+    /// trigger regulary. The period of the timer is left to the implementation
+    /// to decide. The implementation must ensure that it doesn't trigger too
+    /// early (when we haven't hung for example) or too late as to not catch
+    /// faults.
+    fn setup(&self) {}
+
+    /// This function must tickle the watchdog to reset the timer.
+    /// If the watchdog was previously suspended then this should also
+    /// resume the timer.
+    fn tickle(&self) {}
+
+    /// Suspends the watchdog timer. After calling this the timer should not
+    /// fire until after `tickle()` has been called. This function is called
+    /// before sleeping.
+    fn suspend(&self) {}
```

Comment:
Should there possibly be a dedicated `resume` method (perhaps with a default implementation that calls `tickle`?).

I'm not exactly well-versed in the array of watchdog interfaces in the world, but I just want to make sure that we aren't conflating two operations into one interface that would make it hard for some chip to implement -- I think it might also simplify `tickle` implementations on platforms where tickling doesn't auto re-enable, as it would not have to check the status first on every tickle.

Reply:
Good point. I did think about it and thought `tickle()` should be enough, but you are right that maybe on some hardware that could add complexity.

I'll add a `resume()` function